ethexe/contracts/src/Router.sol

```diff
@@ -343,22 +405,21 @@ contract Router is IRouter, OwnableUpgradeable, ReentrancyGuardTransient {
         return keccak256(transitionsHashes);
     }
 
-    function _setValidators(Storage storage router, address[] memory _validators) private {
-        require(router.validationSettings.validators.length == 0, "remove previous validators first");
-
-        for (uint256 i = 0; i < _validators.length; i++) {
-            router.validationSettings.validatorsKeyMap[_validators[i]] = true;
+    function _resetValidators(
+        Gear.Validators storage _validators,
+        address[] memory _newValidators,
+        uint256 _useFromTimestamp
+    ) private {
```

Comment:
it can be very expensive with large number of validators

Reply:
Cheaper than `remove` + `set`.
And also this can be optimised to O(1) in case we would add frost